state/estimator.go

```diff
@@ -55,6 +55,9 @@ func (e *estimator) Stop(context.Context) error {
 	return nil
 }
 
+// estimate the gas price and limit.
+// if gas price is not default and gas limit is set then
+// this 2 values going to be returned (skipping all other checks).
```

Comment:
if gas price and gas limit are both set, then these two values will be returned, bypassing other checks.

Reply:
Done